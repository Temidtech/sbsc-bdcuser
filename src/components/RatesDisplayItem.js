import React, { Component, Fragment } from 'react';
import { View, Text, Image, FlatList } from 'react-native';
import { RateScreenStyles as styles } from '../styles';
import { getImageFromCurrency } from '../dataApi';

export default class Rat extends Component {
  state = {
    rates: this.props.rates,
    date: this.props.date
  }

  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.rates !== prevState.rates) {
      return {
        rates: nextProps.rates,
      };
    }
    return null;
  }

  render() {
    const { rates } = this.state;
    return (
      <Fragment>
        {Object.keys(rates).length ?
          <View>
            {Object.values(rates).map((value, index) =>
              <View key={index}>
                <FlatList
                  data={value}
                  onEndReachedThreshold={0.5}
                  keyExtractor={(item, index) => index.toString()}
                  renderItem={({ item }) => (
                    <View style={[styles.row, styles.margin]}>
                      <View style={[styles.row, styles.date]}>
                        {item.periodOfTheDay === 'MORNING' && <View style={[styles.indicator, styles.morning]}></View>}
                        {item.periodOfTheDay === 'AFTERNOON' && <View style={[styles.indicator, styles.afternoon]}></View>}
                        {item.periodOfTheDay === 'EVENING' && <View style={[styles.indicator, styles.evening]}></View>}
                      </View>
                      <View style={styles.currency}>
                        <Image source={getImageFromCurrency(item.currency)} style={styles.currImage} />
                        <Text style={styles.rateText}>{item.currency}</Text>
                      </View>
                      <View style={styles.sellRate}>
                        <Text style={styles.rateText}>{item.sellRate}</Text>
                      </View>
                      <View style={styles.buyRate}>
                        <Text style={styles.rateText}>{item.buyRate}</Text>
                      </View>
                    </View>
                  )}
                />
              </View>
            )}
          </View> :
          <View style={styles.noRatesWrapper}>
            <Text style={styles.noRatesText}>No rates available for this location</Text>
          </View>}
      </Fragment>
    );
  }
}
