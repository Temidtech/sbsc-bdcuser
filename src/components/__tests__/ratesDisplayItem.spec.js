import React from 'react';
import renderer from 'react-test-renderer';
import RatesDisplayItem from '../RatesDisplayItem';

const rates = {};
describe('<RatesDisplayItem />', () => {

  it('renders the RatesDisplayItem component correctly', () => {
    const tree = renderer.create(
      <RatesDisplayItem rates={rates}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
