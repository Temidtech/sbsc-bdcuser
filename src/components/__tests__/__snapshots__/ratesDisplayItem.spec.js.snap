// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatesDisplayItem /> renders the RatesDisplayItem component correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 150,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 18,
      }
    }
  >
    No rates available for this location
  </Text>
</View>
`;
